src/pudl/settings.py

```diff
@@ -71,64 +71,94 @@ def validate_tables(cls, tables):  # noqa: N805
         return sorted(set(tables))
 
 
-def create_dataset_settings(name: str) -> GenericDatasetSettings:
+class Ferc1Settings(GenericDatasetSettings):
     """
-    Create a pydantic settings class for a dataset.
+    An immutable pydantic model to validate Ferc1Settings.
 
-    This function dynamically creates subclasses of GenericDatasetSettings
-    using information from the constants WORKING_PARTITIONS and PUDL_TABLES.
-    The returned class can be subclassed to add additional fields or validators.
+    Parameters:
+        years: List of years to validate.
+        tables: List of tables to validate.
+
+        working_partitions: Dictionary of working paritions.
+        working_tables: List of working tables.
+    """
+
+    working_partitions: ClassVar = DataSource.from_id("ferc1").working_partitions
```

Comment:
Does it make sense to save this information here separately? Would it be better (or worse!) to just have a FERC 1 `DataSource` that's embedded in the class or the instance, and rather than storing `working_partitions` and `years` have a method that provides access to that information indirectly, so it's only stored in one place (and that place can conceivably change later if it needs to?)

Reply:
I don't see any reason not to store the whole `DataSource`s in the settings. Similar to passing around the whole settings objects vs just the working partitions, this would also give us access to any information in the `DataSource` that may become useful in the future